docs/source/code-generation/codegen-configuration.mdx

```diff
@@ -0,0 +1,359 @@
+		"moduleType": {
+			"swiftPackageManager": {}
+		},
+		"path": "./generated/schema/"
+	},
+	"operations": {
+		"inSchemaModule": {}
+	},
+	"testMocks": {
+		"none": {}
+	},
+	"operationIdentifiersPath": "./generated/"
+}
+```
+
+</ExpansionPanel>
+
+<ExpansionPanel title="Sample Swift">
+
+```swift
+let configuration = ApolloCodegenConfiguration(
+	// Other properties not shown
+	output: ApolloCodegenConfiguration.FileOutput(
+		schemaTypes: ApolloCodegenConfiguration.SchemaTypesFileOutput(
+			path: "./generated/schema/",
+			moduleType: .swiftPackageManager)
+		operations: .inSchemaModule,
+		testMocks: .none,
+		operationIdentifiersPath: "./generated/")
+)
+```
+
+</ExpansionPanel>
+
+#### Schema types
+
+Schema types are common to all generated operation models and are required for the generated Swift code to compile. It is recommended that the schema types be created in a separate module to help with separation of concerns and quicker build times.
+
+The `schemaTypes` property is configured with a `path`, specifying the output location, and a `moduleType` which can help automate module creation. Code generation supports a number of different module types:
+
+| Value | Description |
+| ----- | ----------- |
+| `embeddedInTarget(name:)` | No module will be created and the generated schema types must be manually embedded in a target with the specified `name`. *Note: You are required to add the generated files to your application target.* |
+| `swiftPackageManager` | Generates a `Package.swift` file that is suitable for linking the generated schema types files to your project using Swift Package Manager. |
+| `other` | No module will be created for the generated types and you are required to create the module to support your preferred dependency manager. *Note: You must specify the name of the module you will create in the `schemaName` property as this will be used in `import` statements of generated operation files.* |
+
+#### Operations
+
+Operation files are generated from all the GraphQL operation definitions matched by the `operationSearchPaths` patterns. Each operation (query, mutation, or subscription) will be created in a separate file.
+
+The `operations` property value determines the location of the generated operation files:
+
+| Value | Description |
+| ----- | ----------- |
+| `inSchemaModule` | All operation files will be located in the module with the schema types. |
+| `relative(subpath:)` | Operation object files will be co-located relative to the defining operation file. If `subpath` is specified a subfolder will be created relative to the definition file and the object files will be generated there. If no `subpath` is specified then all operation files will be generated alongside the definition file. |
+| `absolute(path:)` | All operation object files will be located in the specified path. |
+
+#### Test mocks
+
+Test mock objects can be created for your operation models, they can be configured with the following values:
+
```

Comment:
@AnthonyMDev - I feel like we need more detail here but am at a loss of what that is. Would appreciate your input - thanks.

Reply:
I'm going to write a whole page on Test mocks today, so we should link out to that here probably. I'll handle it!